function fetchNumber() {
    return new Promise((resolve, reject) => {
      const num = Math.floor(Math.random() * 10); // Simulating fetching a number
      if (num >= 0) {
        resolve(num);
      } else {
        reject(new Error("Failed to fetch a number"));
      }
    });
  }
fetchNumber()
  .then((num) => {
    console.log("First number:", num);
    return num * 2;
  })
  .then((doubled) => {
    console.log("Doubled value:", doubled);
    return doubled + 5;
  })
  .then((final) => {
    console.log("Final result:", final);
  })
  .catch((err) => {
    console.error("Error:", err.message);
  });
